components/brave_wallet_ui/components/desktop/hardware-wallet-connect/accounts_list.tsx

```diff
@@ -71,48 +58,44 @@ import {
 import { Row } from '../../shared/style'
 
 interface Props {
-  hardwareWallet: HardwareVendor
-  accounts: BraveWallet.HardwareWalletAccount[]
-  preAddedHardwareWalletAccounts: BraveWallet.AccountInfo[]
+  currentHardwareImportScheme: HardwareImportScheme
+  accounts: Array<Required<AccountFromDevice>>
   onLoadMore: () => void
-  selectedDerivationPaths: string[]
-  setSelectedDerivationPaths: (paths: string[]) => void
-  selectedDerivationScheme: string
-  setSelectedDerivationScheme: (scheme: string) => void
+  onAccountChecked: (path: string, checked: boolean) => void
+  setHardwareImportScheme: (scheme: DerivationScheme) => void
   onAddAccounts: () => void
-  filecoinNetwork: FilecoinNetwork
-  onChangeFilecoinNetwork: (network: FilecoinNetwork) => void
-  coin: BraveWallet.CoinType
+}
+
+const defaultNetworkId = (coin: BraveWallet.CoinType) => {
+  if (coin === BraveWallet.CoinType.ETH) return BraveWallet.MAINNET_CHAIN_ID
+  if (coin === BraveWallet.CoinType.SOL) return BraveWallet.SOLANA_MAINNET
+  if (coin === BraveWallet.CoinType.BTC) return BraveWallet.BITCOIN_MAINNET
+  if (coin === BraveWallet.CoinType.FIL) return BraveWallet.FILECOIN_MAINNET
+
+  assertNotReached(`Unknown coin ${coin}`)
 }
 
 export const HardwareWalletAccountsList = ({
+  currentHardwareImportScheme,
+  setHardwareImportScheme,
   accounts,
-  preAddedHardwareWalletAccounts,
-  hardwareWallet,
-  selectedDerivationScheme,
-  setSelectedDerivationScheme,
-  setSelectedDerivationPaths,
-  selectedDerivationPaths,
   onLoadMore,
-  onAddAccounts,
-  filecoinNetwork,
-  onChangeFilecoinNetwork,
-  coin
+  onAccountChecked,
+  onAddAccounts
 }: Props) => {
+  const coin = currentHardwareImportScheme.coin
+  const hardwareVendor = currentHardwareImportScheme.vendor
```

Comment:
nit: can we use de-structuring instead?
```suggestion
  const { coin, vendor: hardwareVendor } = currentHardwareImportScheme
```